tests/chainer_tests/training_tests/extensions_tests/test_nan_killer.py

```diff
@@ -0,0 +1,84 @@
+import os
+import tempfile
+import unittest
+
+import numpy
+
+import chainer
+from chainer import links
+from chainer import testing
+from chainer.testing import attr
+from chainer import training
+
+
+class Model(chainer.Chain):
+
+    def __init__(self):
+        super(Model, self).__init__()
+        with self.init_scope():
+            self.l = links.Linear(1, 3)
+
+    def __call__(self, x):
+        return self.l(x)
+
+
+class Dataset(chainer.dataset.DatasetMixin):
+
+    def __init__(self, values):
+        self.values = values
+
+    def __len__(self):
+        return len(self.values)
+
+    def get_example(self, i):
+        return numpy.array([self.values[i]], numpy.float32), numpy.int32(i % 2)
+
+
+class TestNaNKiller(unittest.TestCase):
+
+    def setUp(self):
+        self.n_data = 4
+        self.n_epochs = 3
+
+        self.model = Model()
+        self.classifier = links.Classifier(self.model)
+        self.optimizer = chainer.optimizers.Adam()
+        self.optimizer.setup(self.classifier)
+
+        self.dataset = Dataset([i for i in range(self.n_data)])
+        self.iterator = chainer.iterators.SerialIterator(
+            self.dataset, 1, shuffle=False)
+
+    def prepare(self, device=None):
+        tempdir = tempfile.mkdtemp()
```

Comment:
Please cleanup temp directory used for tests.
How about creating it at the last step of `setUp` and use `tearDown` method to remove it?